src/trio/_core/_run.py

```diff
@@ -540,11 +540,21 @@ class CancelScope:
     _has_been_entered: bool = attrs.field(default=False, init=False)
     _registered_deadline: float = attrs.field(default=inf, init=False)
     _cancel_called: bool = attrs.field(default=False, init=False)
+    _is_relative: bool | None = attrs.field(default=False, init=False)
```

Comment:
I think we can actually avoid tracking this attribute at all, and instead define:

```python
@property
def is_relative(self):
    if self.deadline < inf and not self._has_been_entered:
        assert self.relative_deadline == inf
        return True
    if self.relative_deadline < inf and not self._has_been_entered:
        return False
    return None
```

Reply:
nice, that works.
Re `< inf` vs `!= inf`: we made the timeout helper functions have special handling for `nan`, but not `CancelScope`... oops